### PR TITLE
MISC: use pnpm

### PR DESCRIPTION
**Pull request by dannypsnl**

### CODE REVIEW

**Comment by re-xyr** [comment]
The build script is still using `yarn`. Could you change it to use `pnpm` too?

**Comment by dannypsnl** [comment]
Get it

**Comment by dannypsnl** [comment]
It seems like we need to install pnpm back to upstream(npm cannot read installation by pnpm thus my fixed won't work

**Comment by dannypsnl** [comment]
@re-xyr I have done the patch to lannonbr/vsce-action@master, could you approve CI?

**Comment by ice1000** [comment]
It should be running

**Comment by ice1000** [comment]
Failed

**Comment by dannypsnl** [comment]
> Failed

I see, stupid error. package.json be reverted but I didn't update it again.

**Comment by dannypsnl** [comment]
NOTE: to fix 

ERROR: This version of pnpm requires at least Node.js v12.17
11
The current version of Node.js is v10.24.1

**Comment by ice1000** [comment]
Do we really need these much dependencies? I wonder if we can simplify the lock files

**Comment by re-xyr** [comment]
It is the vscode language client that needs that many dependencies, and we depend on the language client. And it is better to lock all the dependencies’ versions to have reproducible builds.

**Comment by ice1000** [comment]
Ok

**Comment by ice1000** [comment]
Fuck JavaScript ecosystem

**Comment by re-xyr** [comment]
Speaking of the lock files, I believe Java build systems forces you to have one exact version for a dependency so you don’t need lock files. Haskell stack is crazier, for each LTS ghc version it has a giant lock file that locks all packages versions stored in stackage, and this giant lock file is referred to by an LTS version number, so you don’t need lock files for each of your project.

**Comment by dannypsnl** [comment]
need to rerun, I use `node:12-slim` in `dannypsnl-fork/vsce-action` now

**Comment by ice1000** [comment]
You can push an empty commit. I decide to squash this so do whatever you want!

**Comment by dannypsnl** [comment]
> You can push an empty commit. I decide to squash this so do whatever you want!

updated, CI

**Comment by dannypsnl** [comment]
try install vsce by pnpm, CI

File: .github/workflows/vsix.yml [hunk 1]
```diff
@@ -2,9 +2,9 @@ name: build vsix
 
 on:
   push:
-    branches: [ main ]
+    branches: [main]
   pull_request:
-    branches: [ main ]
+    branches: [main]
```
Comment (ice1000) [inline .github/workflows/vsix.yml:7]: Are these changes necessary?

File: .github/workflows/vsix.yml [hunk 2]
```diff
@@ -15,18 +15,17 @@ jobs:
...
-    - run: yarn install
-    - uses: lannonbr/vsce-action@master
-      with:
-        args: "package"
-    - uses: actions/upload-artifact@v2
-      with:
-        name: aya-prover-vscode-extension
-        path: ./aya-prover-vscode-*.vsix
-
+      - uses: actions/checkout@v2
```
Comment (ice1000) [inline .github/workflows/vsix.yml:18]: Is this new indentation necessary?

**Comment by ice1000** [comment]
Check failed

**Comment by dannypsnl** [comment]
```
> aya-prover-vscode@0.0.1 compile /github/workspace
> tsc -p ./

 ERROR  Command failed: npm list --production --parseable --depth=99999 --loglevel=error
npm ERR! missing: minimatch@^3.0.4, required by vscode-languageclient@7.0.0
npm ERR! missing: semver@^7.3.4, required by vscode-languageclient@7.0.0
npm ERR! missing: vscode-languageserver-protocol@3.16.0, required by vscode-languageclient@7.0.0
```

emmm....but there shall not have any call to `npm`.

File: .github/workflows/vsix.yml (hunk 1 above)
Comment (dannypsnl) [reply .github/workflows/vsix.yml:7]: no, maybe I can close auto formatter

**Comment by ice1000** [comment]
😥
